### PR TITLE
Try Architecture independent Build

### PR DESCRIPTION
**Pull request by specter119**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Review by xhochy — CHANGES_REQUESTED** [review]
noarch and compiled code don't work together.

**Comment by specter119** [comment]
@xhochy I appreciate your review and your feedback.
I have examined the Conda packages and PyPI whl, and I did not find any `.so`, `.a`, `.dll`, or `.pyd` like files. The `.pyc` files should be generated only when the code is executed. `shap` seems to be a pure python package at the moment.

Based on these observations, and the fact that `shap` is not actively developed, I think changing this package to noarch could benefit the packages that depend on `shap` when a new Python minor version is released. It could also reduce your maintenance burden.

Thank you for your response, and I respectfully request that you take another look at your decision with these points in mind.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -27,7 +28,7 @@ requirements:
     - setuptools
     - numpy
   run:
-    - python
+    - python >=3.6,<4.0
```
Comment (xhochy) [inline recipe/meta.yaml:31]: Please remove the upper bound, we try to keep this open.
```suggestion
    - python >=3.6
```

**Review by xhochy — CHANGES_REQUESTED** [review]

**Comment by xhochy** [comment]
From the build log:

```
   INFO: sysroot: '/home/conda/feedstock_root/build_artifacts/shap_1684470949380/_build_env/x86_64-conda-linux-gnu/sysroot/' files: '['/home/conda/feedstock_root/build_artifacts/shap_1684470949380/_build_env/x86_64-conda-linux-gnu/sysroot/usr/share/zoneinfo/zone.tab', '/home/conda/feedstock_root/build_artifacts/shap_1684470949380/_build_env/x86_64-conda-linux-gnu/sysroot/usr/share/zoneinfo/tzdata.zi', '/home/conda/feedstock_root/build_artifacts/shap_1684470949380/_build_env/x86_64-conda-linux-gnu/sysroot/usr/share/zoneinfo/right/Zulu', '/home/conda/feedstock_root/build_artifacts/shap_1684470949380/_build_env/x86_64-conda-linux-gnu/sysroot/usr/share/zoneinfo/right/WET']'
WARNING (shap,lib/python3.10/site-packages/shap/_cext.cpython-310-x86_64-linux-gnu.so): Needed DSO lib/libstdc++.so.6 found in ['libstdcxx-ng']
WARNING (shap,lib/python3.10/site-packages/shap/_cext.cpython-310-x86_64-linux-gnu.so): .. but ['libstdcxx-ng'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)
   INFO (shap,lib/python3.10/site-packages/shap/_cext.cpython-310-x86_64-linux-gnu.so): Needed DSO x86_64-conda-linux-gnu/sysroot/lib64/libm.so.6 found in CDT/compiler package conda-forge::sysroot_linux-64-2.12-he073ed8_15
   INFO (shap,lib/python3.10/site-packages/shap/_cext.cpython-310-x86_64-linux-gnu.so)
```

This is an architecture-specific package.

**Comment by specter119** [comment]
@xhochy I will try to further check it, thx.

**Comment by specter119** [comment]
@xhochy yes, I found the .so and .pyd files are in the root folder, so this cannot be architecture-specific, thx

**Pull request closed without merging**